src/runner.rs

```diff
@@ -78,8 +77,11 @@ impl<E: EnvController> Runner<E> {
             };
             let db = self.env_controller.start(&env, config_path).await;
             if let Err(e) = self.run_env(&env, &db).await {
-                println!("Environment {} run failed with error {:?}", env, e);
+                println!("Environment {} run failed, error:{:?}.", env, e);
+                self.env_controller.stop(&env, db).await;
+                return Err(e);
```

Comment:
Error handling here should also consider the `fail_fast` option. If not fail fast, it should keep running the next env and collect errors after all then return.

Reply:
Make sense, will fix this.